robottelo/ui/session.py

```diff
@@ -1,55 +1,245 @@
+        self.medium = Medium(self.browser)
+        self.my_account = MyAccount(self.browser)
+        self.navigator = Navigator(self.browser)
+        self.nav = self.navigator  # for compatibility purposes
+        self.user = User(self.browser)
+        self.operatingsys = OperatingSys(self.browser)
+        self.org = Org(self.browser)
+        self.oscapcontent = OpenScapContent(self.browser)
+        self.oscappolicy = OpenScapPolicy(self.browser)
+        self.oscapreports = OpenScapReports(self.browser)
+        self.package = Package(self.browser)
+        self.partitiontable = PartitionTable(self.browser)
+        self.puppetclasses = PuppetClasses(self.browser)
+        self.puppetmodule = PuppetModule(self.browser)
+        self.products = Products(self.browser)
+        self.registry = Registry(self.browser)
+        self.repository = Repos(self.browser)
+        self.rhai = RHAI(self.browser)
+        self.role = Role(self.browser)
+        self.settings = Settings(self.browser)
+        self.sc_parameters = SmartClassParameter(self.browser)
+        self.smart_variable = SmartVariable(self.browser)
+        self.subnet = Subnet(self.browser)
+        self.subscriptions = Subscriptions(self.browser)
+        self.sync = Sync(self.browser)
+        self.syncplan = Syncplan(self.browser)
+        self.task = Task(self.browser)
+        self.template = Template(self.browser)
+        self.trend = Trend(self.browser)
+        self.usergroup = UserGroup(self.browser)
+
+        # for compatibility purposes
+        for attr in (
+                'activationkey', 'architecture', 'bookmark', 'container',
+                'compute_profile', 'compute_resource', 'contenthost',
+                'configgroups', 'content_views', 'dashboard', 'dockertag',
+                'domain', 'errata', 'discoveredhosts', 'discoveryrules',
+                'environment', 'gpgkey', 'hardwaremodel', 'hostcollection',
+                'hostgroup', 'hosts', 'job', 'jobtemplate', 'ldapauthsource',
+                'lifecycleenvironment', 'location', 'login', 'medium',
+                'my_account', 'navigator', 'nav', 'user', 'operatingsys',
+                'org', 'oscapcontent', 'oscappolicy', 'oscapreports',
+                'package', 'partitiontable', 'puppetclasses', 'puppetmodule',
+                'products', 'registry', 'repository', 'rhai', 'role',
+                'settings', 'sc_parameters', 'smart_variable', 'subnet',
+                'subscriptions', 'sync', 'syncplan', 'task', 'template',
+                'trend', 'usergroup'):
+            setattr(self.test, attr, getattr(self, attr))
+
+        self.login.login(self._user, self._password)
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if exc_type is None:
-            self.logout()
+        try:
+            if exc_type is None:
+                self.login.logout()
+            else:
+                if settings.browser != 'saucelabs':
```

Comment:
your are going to start your investigation anyway from jenkins and go on saucelab afterward. I think brief information and one screenshot should be enough to analyze 90% of failures, so probably let's keep some information on Jenkins. Just my opinion, nothing more